ui: mobile player (0.7)

espeon opened the pull request.

espeon committed:
Add convenience hooks

espeon committed:
move more convenience hooks

espeon committed:
export the hooks

espeon committed:
Add stream record creation and start streamer context menu

espeon committed:
Make UI and chat UI more streamlined

espeon committed:
clean up imports and random comments

espeon committed:
Add docs

iameli committed:
components: add some relative paths

espeon committed:
add streamplace-desktop to stream key detection

espeon committed:
Rework chat rendering

espeon committed:
Add reply preview to chat message component

espeon committed:
Fix imports

espeon committed:
Don't warn on webrtc on 127.0.0.1

espeon committed:
Improve chat reply UI and fix message ordering

iameli commented:
@espeon CI error is a little opaque but I think this means these packages are used in the library without being declared as package.json dependencies 
![image](https://github.com/user-attachments/assets/ce33d960-e5b3-445a-bc54-20377abaec70)

espeon committed:
Add swipe-to-reply gesture to chat messages on New Chat on all platforms
except android

espeon committed:
Add swipe-right mod actions to chat messages

espeon committed:
misc chat style improvements

espeon committed:
Make chat box resizable

espeon committed:
Change objectFit from cover to contain in RTCViewIngest

espeon committed:
Add components

espeon committed:
Add chat box with mention suggestions and reply

espeon committed:
Simplify chat layout a tad

espeon committed:
small nits

- hide mobile go live on web - change back button to white to contrast
better with bg

iameli commented on js/docs/src/content/docs/components/custom_ui.md:
```diff
@@ -0,0 +1,245 @@
+  How to set up your player UI with components from @streamplace/components.
+---
+
+# Building a Custom Player UI
```
title is already there 
```suggestion
```

iameli reviewed (approved):
Looking really great! Tracking some stuff to fix but let's get this merged in and do smaller PRs on top of it, this is getting big
